tests/agent/actions/test_draft_datasource_config.py

```diff
@@ -0,0 +1,169 @@
+import os
+from unittest.mock import MagicMock
+from uuid import UUID
+
+import pytest
+
+from great_expectations.agent.actions.test_draft_datasource_config import (
+    DraftDatasourceConfigAction,
+)
+from great_expectations.agent.agent import GxAgentEnvVars
+from great_expectations.agent.models import DraftDatasourceConfig
+from great_expectations.data_context import CloudDataContext
+from great_expectations.datasource.fluent import SQLDatasource
+from great_expectations.datasource.fluent.interfaces import TestConnectionError
+
+
+@pytest.fixture(scope="function")
+def context():
+    return MagicMock(autospec=CloudDataContext)
+
+
+@pytest.fixture
+def org_id():
+    return "81f4e105-e37d-4168-85a0-2526943f9956"
+
+
+@pytest.fixture
+def token():
+    return "MTg0NDkyYmYtNTBiOS00ZDc1LTk3MmMtYjQ0M2NhZDA2NjJk"
+
+
+@pytest.fixture(autouse=True)
```

Comment:
```suggestion
@pytest.fixture
```
I think we should avoid `autouse` unless absolutely necessary, it makes things harder to reason about.

Reply:
yep, we normally don't use `autouse`, but in this case every single test in this module requires this fixture and has no use for the return value within the test, so this is the correct time to use it.

Reply:
You can use it in your `context` fixture and still get rid of `autouse`

```python

@pytest.fixture(scope="function")
def context(set_required_env_vars):
    return MagicMock(autospec=CloudDataContext)

...

def test_test_draft_datasource_config_success(context, mocker):
   ...
```

Reply:
okay, looks like we just have different opinions here. Why would `context` require that fixture? it doesn't need the environment variables, and has no use for the return value. When we write a test that defines its own `context` mock, that test will unexpectedly fail due to missing environment variables, even though its otherwise identical to other tests in the module. That seems more magical and more unexpected.

Reply:
I agree It doesn't need the fixture.
I would prefer if all the tests were explicit, about the fixtures they use. The example with `context`, was to illustrate how fixtures can use other fixtures.

```python
def test_test_draft_datasource_config_success(context, mocker, set_required_env_vars):
   ...
```

Reply:
I agree with avoiding `autouse` if possible! Explicit is always better than implicit and as a future reader of this test, it would be nice to know exactly what setup logic is occurring before I get to the actual core logic of the test. 

I'm of the opinion that `pytest` is already a little too magical so it would be nice to minimize that wherever we can imo.

Reply:
+1 on preferring minimal use of autouse (as someone who is regretful author of several autouse fixtures). As useful as I've found that option in the past, now that we have a lot more tests, I'm learning how much it can bog down test debugging. Particularly with pytest's complicated fixture import/scope hierarchy. And especially when working in a new part of codebase.

Reply:
updated in [62ef383](https://github.com/great-expectations/great_expectations/pull/8410/commits/62ef3838fd686b8bd586649aa9c8483af59bdae2)

Reply:
if this is a blocking req, this needs to be added to our style guide

Reply:
I think we're adding this as an agenda item for OSS arch review tomorrow! Don't think this should be a blocker but a good conversation to have